pkgs/development/libraries/harfbuzz/default.nix

```diff
@@ -58,9 +74,10 @@ stdenv.mkDerivation {
   ];
 
   buildInputs = [ glib freetype cairo ] # recommended by upstream
-    ++ lib.optionals withCoreText [ ApplicationServices CoreText ];
+    ++ lib.optionals withCoreText [ ApplicationServices CoreText ]
+    ++ lib.optionals (stdenv.buildPlatform == stdenv.hostPlatform) [ gobject-introspection ];
```

Comment:
```suggestion
    ++ lib.optionals withIntrospection [ gobject-introspection ];
```

Reply:
It seems that you and @jtojnar disagree on the style here. What shall I do here? I personally find the `withIntrospection` nice since it's also easy to search for if the introspection stuff gets updated to support cross.

But I don't have a strong opinion on what's best.

Reply:
Yeah, I am trying to reduce the noise in the expressions I maintain – the chance that someone will want to disable introspection aside from cross-compiling is pretty small so the additional clutter is not worth it IMO. But maintainers of each package will have to choose.

Reply:
@jtojnar says let bind it rather than make it a parameter. I don't care strongly about that; I just want a variable to be somehow bound (let or parameter) with a descriptive name, rather than repeating the cross check which isn't inherently related to introspection.

Reply:
Thanks for the tip! Updated with a let-expression.